src/Persistence/Mapping/AbstractClassMetadataFactory.php

```diff
@@ -153,6 +153,24 @@ abstract protected function initializeReflection(
      */
     abstract protected function isEntity(ClassMetadata $class);
 
+    /**
+     * Removes the prepended backslash of a class string to conform with how php outputs class names
+     *
+     * @psalm-param string $className
+     *
+     * @psalm-return class-string
+     *
+     * @psalm-suppress MoreSpecificReturnType
```

Comment:
Isn't that for global suppression only? I don't see how to suppress only for a method in that file

Reply:
I guess I could try adding a stub file as a patch since I managed to fix this issue in psalm, the stub file would just need this

```
/**
 * @psalm-pure
 *
 * @return ($string is class-string ? ($characters is '\\' ? class-string : string) : string)
 *
 * @psalm-flow ($string) -> return
 */
function ltrim(string $string, string $characters = " \t\n\r\0\x0B") : string {}
```